### PR TITLE
feat: added comment to report reasons

### PR DESCRIPTION
**Pull request by rebelchris**
When a user reports a post, they can now provide a comment with their chosen reason.

DD-40

### CODE REVIEW

File: __tests__/posts.ts [hunk 1]
```diff
@@ -734,8 +734,8 @@ describe('mutation banPost', () => {
 
 describe('mutation reportPost', () => {
```
Comment (idoshamun) [inline __tests__/posts.ts:735]: Can you add a test that doesn't send a comment so we'll check for backward compatibility?

File: src/entity/PostReport.ts [hunk 2]
```diff
@@ -17,6 +17,9 @@ export class PostReport {
   @Column({ length: 12 })
   reason: string;
 
+  @Column({ type: 'text' })
+  comment: string;
```
Comment (idoshamun) [inline src/entity/PostReport.ts:21]:
```suggestion
  @Column({ type: 'text', nullable: true })
  comment?: string;
```

**Review by idoshamun — CHANGES_REQUESTED** [review]
Please also add the new comment field to Slack's notification.

**Comment by rebelchris** [comment]
@idoshamun Thanks for spotting the slack webhook.
Also added test for empty comments.

**Review by idoshamun — CHANGES_REQUESTED** [review]
I think you forgot to regenerate the migration

**Review by sshanzel — CHANGES_REQUESTED** [review]
Just this one super tiny thing with linting.

File: src/schema/posts.ts [hunk 3]
```diff
@@ -494,9 +502,12 @@ export const resolvers: IResolvers<any, Context> = {
...
         if (!post.banned) {
           try {
-            await ctx
-              .getRepository(PostReport)
-              .insert({ postId: id, userId: ctx.userId, reason: reason });
+            await ctx.getRepository(PostReport).insert({
+              postId: id,
+              userId: ctx.userId,
+              reason: reason,
+              comment: comment,
```
[the remark this answers is not in the document]
Comment (rebelchris) [reply src/schema/posts.ts]: Oh nice spot! Thanks Lee 🎉

**Review by sshanzel — APPROVED** [review]
Woohoo! LGTM! 🚢

**Comment by rebelchris** [comment]
@idoshamun Would you mind checking this one as well.

**Review by idoshamun — APPROVED** [review]
LGTM!
